mass_mailing_partner/views/res_partner_view.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
+<openerp>
+    <data>
+
+        <record id="act_partner_mass_mailing_contacts" model="ir.actions.act_window">
+            <field name="res_model">mail.mass_mailing.contact</field>
+            <field name="view_type">form</field>
+            <field name="name">Mailing lists</field>
+            <field name="view_mode">tree,form,calendar,graph</field>
+            <field name="context">{'search_default_partner_id': active_id, 'default_partner_id': active_id}</field>
+            <field name="domain">[('partner_id', '=', active_id)]</field>
+        </record>
+
+        <record model="ir.ui.view" id="view_partner_form_mass_mailing_contacts">
+            <field name="name">Partner Form with mailing contacts</field>
+            <field name="model">res.partner</field>
+            <field name="inherit_id" ref="base.view_partner_form"/>
+            <field name="arch" type="xml">
+                <div class="oe_right oe_button_box" position="inside">
+                    <button name="%(mass_mailing_partner.act_partner_mass_mailing_contacts)d"
```

Comment:
You can use an existing act_window and save the declaration, translation and so on. You only need to put the same context you're using in the other act_window:

```
<button name="%(mass_mailing.action_view_mass_mailing_contacts)d"
        context="{'search_default_partner_id': active_id, 'default_partner_id': active_id}"
```

The only drawback is that the domain can be preserved, but I even prefer this option, because the filter is applied anyway, and you have the option to remove the filter and see all contacts without changing the screen.